server/app/rest_api/rest.py

```diff
@@ -201,6 +201,53 @@ def get(self):
         else:
             return make_response(jsonify(annotation_response))
 
+    @swagger.doc({
+        "summary": "Fetch annotations (metadata) for filtered subset.",
+        "tags": ["annotations"],
+        "parameters": [
+            {
+                "in": "query",
+                "name": "annotation-name",
+                "type": "string",
+                "description": "comma-separated annotation keys, ex: num_genes,percent_mito"
+            },
+            {
+                'name': 'filter',
+                'description': 'Complex Filter',
+                'in': 'body',
+                'schema': FilterModel
+            }
+        ],
+        "responses": {
+            "200": {
+                "description": "annotations",
+                "examples": {
+                    "application/json": {
+                        "names": [
+                            'tissue_type', 'sex', 'num_reads', 'clusters'
+                        ],
+                        "data": [
+                            [0, 'lung', 'F', 39844, 99],
+                            [1, 'heart', 'M', 83, 1],
+                            [49, 'spleen', None, 2, "unknown cluster"],
+
+                        ]
+                    }
+
+                }
+            }
+        }
+    })
+    def put(self):
+        fields = request.args.getlist("annotation-name", None)
+        df = current_app.data.filter_dataframe(request.get_json()["filter"])
+        try:
+            annotation_response = current_app.data.annotation(df, fields)
+        except KeyError:
+            return make_response(f"Error bad key in {fields}", 404)
+        else:
```

Comment:
technically, since you return on all the exception handling paths, you don't need the else.

Reply:
fixed